raiden/network/sockfactory.py

```diff
@@ -0,0 +1,59 @@
+from contextlib import contextmanager
+from raiden.network import upnpsock, stunsock
+
+
+class PortMappedSocket(object):
+    """Wrapper around a socket instance with port mapping information.
+    """
+    def __init__(self, sock, method, external_ip, external_port, **meta):
+        self.socket = sock
+        self.method = method
+        self.external_ip = external_ip
+        self.external_port = external_port
+        self.meta = meta
+
+    def __repr__(self):
+        return 'PortMappedSocket<{} mapping: {}:{} ({})>'.format(
+            self.socket.__repr__(),
+            self.external_ip,
+            self.external_port,
+            self.method
+        )
+
+
+@contextmanager
```

Comment:
The context manager won't execute the code after yield if an exception is raised, please add a try-finally to properly cleanup and re-raise the exception (if one was raised).